java/adapter/jdbc/src/main/java/org/apache/arrow/adapter/jdbc/JdbcToArrowUtils.java

```diff
@@ -265,7 +265,7 @@ public static ArrowType getArrowTypeForJdbcField(JdbcFieldInfo fieldInfo, Calend
       case Types.CLOB:
         return new ArrowType.Utf8();
       case Types.DATE:
-        return new ArrowType.Date(DateUnit.MILLISECOND);
+        return new ArrowType.Date(DateUnit.DAY);
```

Comment:
should also change [line 213](https://github.com/apache/arrow/pull/5944/files#diff-ae45345b37c543f89306497e1ceeafe2R213)?

Reply:
Nice catch. Thank you.